ehrapy/tools/_utils.py

```diff
@@ -0,0 +1,244 @@
+from typing import Iterable, Literal, Union
+
+import numpy as np
+import pandas as pd
+
+from ehrapy.tools import _datatypes
+
+
+def _merge_arrays(arrays: Iterable[Iterable], groups_order):
```

Comment:
```suggestion
def _merge_recarray_with_ndarray(arrays: Iterable[Iterable], groups_order):
```

Maybe something like this? Currently the name is very general.